src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -968,15 +986,31 @@ update_estimate(TargetT &current_alpha_coefficent_image)
   timerSubset.Start();
 #endif // PARALLEL
   
+  shared_ptr<TargetT> iterative_kernel_image_sptr;
+  
   // TODO make member parameter to avoid reallocation all the time
   unique_ptr< TargetT > multiplicative_update_image_ptr
     (current_alpha_coefficent_image.get_empty_copy());
 
   const int subset_num=this->get_subset_num();  
   info(boost::format("Now processing subset #: %1%") % subset_num);
 
+  
+  if (this->stop_iterative_kernel_at_iter_num>0 &&
+      this->subiteration_num%this->get_num_subsets()==0 &&
+      this->subiteration_num/this->get_num_subsets()==this->stop_iterative_kernel_at_iter_num)
```

Comment:
I'm a bit confused here with the two `if`s, can you add some comments to explain the logic? i.e., why do we need this `if` and `iterative_kernel_image_sptr`? Can't we replace the two sets of `if`s here with

```
  if (this->stop_iterative_kernel_at_iter_num<=0 ||
       this->subiteration_num/this->get_num_subsets()<this->stop_iterative_kernel_at_iter_num) {
    iterative_kernel_image_sptr=shared_ptr<TargetT>(current_alpha_coefficent_image.clone());
}
```
I believe this would stop updating iterative kernel_image_sptr when `this->subiteration_num%this->get_num_subsets()==0` by default.

Reply:
Sure definitely the comment needs to be added. 

the first condition is used to set the member  "iterative_kernel_image_fixed_sptr"  to the iteration we want while the second defines "iterative_kernel_image_sptr" which is used when we call "compute_kernel...()".  This can be = to the current update if we did not set the keyword "stop..." or if the current iteration is < stop_kernel... otherwise  iterative_kernel_image_sptr= iterative_kernel_image_fixed_sptr

this->subiteration_num%this->get_num_subsets()==0  this condition is not enough because this would stop the kernel at every full iteration